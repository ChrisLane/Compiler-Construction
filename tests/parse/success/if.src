// This tests the 'if' expression. This should parse.
main() {
    var x = 10;
    if (0==0) {
        return $x;
    } else {
        x = x+1;
        return $x;
    };
}
