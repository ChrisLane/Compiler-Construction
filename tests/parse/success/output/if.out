
main (  ) { 
    New ( "x" = Const 10 ) In { 
        If ( Operator ( Equal, Const 0 , Const 0 ) ) { 
            Return ( Deref ( Identifier "x" ) ) 
        } Else { 
            Asg ( Identifier "x" := Operator ( Plus, Identifier "x" , Const 1 ) ) ; 
            Return ( Deref ( Identifier "x" ) ) 
        } 
    } 
}

